File: dev/cloud.rb
#!/usr/bin/env ruby
require 'atmosphere'

class Cloud
  TYPES = ['Cumulus', 'Stratus', 'Cirrus', 'Nimbostratus'].freeze

  attr_accessor :color, :altitude
  attr_reader :type

  def initialize(color, type = nil)
    # One dark but not so stormy night...
    @color = color
    @type = type || TYPES.sample
    @altitude = gen_altitude
  end

  def climb(increase = nil)
    @altitude += increase if increase
    @altitude
  end

  private

  def gen_altitude
    rand(100..1000)
  end
end

my_cloud = Cloud.new('gray')
print "New cloud: #{my_cloud}\n"

File: dev/cloud.py
#!/usr/bin/env python
import random
from typing import ClassVar


class Cloud:
    TYPES: ClassVar[list[int]] = ["Cumulus", "Stratus", "Cirrus", "Nimbostratus"]

    def __init__(self, color: str, type: str | None = None):
        """One dark but not so stormy night..."""
        self.color = color
        self.type = type or random.choice(self.TYPES)
        self._altitude = None

    @property
    def altitude(self) -> int:
        if self._altitude is None:
            self.altitude = self._gen_altitude()
        return self._altitude

    def climb(self, increase: int | None) -> int:
        if increase:
            self.altitude += increase
        return self.altitude

    def _gen_altitude(self) -> int:
        return random.randint(100, 1000)


my_cloud = Cloud(color="gray")
print(f"New cloud: {my_cloud}\n")
